### PR TITLE
Bring back #36396

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: tests/integration/test_materialized_mysql_database/materialize_with_ddl.py [hunk 1]
```diff
@@ -1796,7 +1796,7 @@ def mysql_settings_test(clickhouse_node, mysql_node, service_name):
         clickhouse_node.query(
             "SELECT COUNT(DISTINCT  blockNumber()) FROM test_database.a FORMAT TSV"
         )
-        == "2\n"
+        == "1\n"
```
Comment (tavplubix) [inline tests/integration/test_materialized_mysql_database/materialize_with_ddl.py:1799]: Hm, but why the result has changed?

Comment (tavplubix) [reply tests/integration/test_materialized_mysql_database/materialize_with_ddl.py:1799]: Sorry, I didn't notice that it selects `blockNumber()`, so test relies on implementation details

**Comment by tavplubix** [comment]
Btw, the release branch for 22.6 is already created and there's no rush, so it makes sense to address review comments from #36396 in this PR.

**Comment by nickitat** [comment]
@Mergifyio update

**Review by rschu1ze — APPROVED** [review]

File: tests/queries/0_stateless/01861_explain_pipeline.reference [hunk 2]
```diff
@@ -16,8 +16,15 @@ ExpressionTransform
 ExpressionTransform × 2
   (ReadFromMergeTree)
   ExpressionTransform × 2
-    ReplacingSorted × 2 2 → 1
-      Copy × 2 1 → 2
-        AddingSelector × 2
-          ExpressionTransform × 2
-            MergeTreeInOrder × 2 0 → 1
+    ReplacingSorted
```
[the remark this answers is not in the document]
Comment (nickitat) [reply tests/queries/0_stateless/01861_explain_pipeline.reference:19]: no, explain pipeline for some reason doesn't recognize that they are parallel, not consecutive transforms.

**Comment by nickitat** [comment]
> @nickitat, Stress tests failure was interesting: [#25225 (comment)](https://github.com/ClickHouse/ClickHouse/issues/25225#issuecomment-1163083897)

unfortunately, I failed to found the reason

**Pull request merged**